iceoryx2-services/discovery/src/service_discovery/service.rs

```diff
@@ -304,12 +355,23 @@ impl<S: ServiceType> Service<S> {
         if discovery_config.sync_on_initialization {
             tracker.sync(iceoryx_config)?;
         }
+        let server = Some(
+            node.service_builder(service_name())
+                .request_response::<(), [StaticConfig]>()
+                .open_or_create()
+                .map_err(|_| CreationError::ServiceCreationFailure)?
+                .server_builder()
```

Comment:
```suggestion
                .server_builder()
                .initial_max_slice_length(discovery_config.initial_max_response_size) // this is a guess for the maximum "number of `StaticConfig` elements to expect, you can choose another name for the config if you like
```

By exposing the `initial_max_slice_length` a user can provide an initial guess for the maximum response size. If the guess is exceeded, more memory is allocated - this takes time, but with a good initial guess should happen less often and the performance should be better.